File: veo_test.mjs
import { GoogleAuth } from 'google-auth-library';
import fs from 'fs';
import fetch from 'node-fetch';

const auth = new GoogleAuth({
  keyFile: '/home/gateman/sa-key.json',
  scopes: ['https://www.googleapis.com/auth/cloud-platform'],
});

async function main() {
  try {
    const client = await auth.getClient();
    const token = await client.getAccessToken();
    const projectId = 'jason-hsbc';
    const location = 'us-central1';
    
    const url = `https://${location}-aiplatform.googleapis.com/v1beta1/projects/${projectId}/locations/${location}/publishers/google/models/veo-3.1-generate-preview:predictLongRunning`;
    
    const payload = {
      instances: [
        { prompt: "A gorgeous and highly elegant female secretary walking gracefully in a modern sunlit office, cinematic lighting, ultra high quality, photorealistic, 4k resolution" }
      ],
      parameters: {
        aspectRatio: "16:9",
        personGeneration: "ALLOW_ADULT"
      }
    };

    console.log("Submitting to Veo 3.1 Preview:", url);
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${token.token}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(payload)
    });
    
    if (!response.ok) {
        console.error("API Error Response:", await response.text());
        return;
    }
    const data = await response.json();
    console.log("Initial Response:", JSON.stringify(data, null, 2));

    if (data.name) {
        console.log("Operation started! ID:", data.name);
        let opName = data.name;
        let isDone = false;
        let attempts = 0;
        
        while (!isDone && attempts < 30) {
            attempts++;
            console.log(`Waiting 10s... (Attempt ${attempts})`);
            await new Promise(r => setTimeout(r, 10000));
            
            const pollUrl = `https://${location}-aiplatform.googleapis.com/v1beta1/${opName}`;
            const pollRes = await fetch(pollUrl, {
                headers: { 'Authorization': `Bearer ${token.token}` }
            });
            if (!pollRes.ok) {
                console.error("Poll Error:", await pollRes.text());
                continue;
            }
            
            const pollData = await pollRes.json();
            if (pollData.done) {
                isDone = true;
                if (pollData.error) {
                    console.error("Operation failed:", JSON.stringify(pollData.error, null, 2));
                } else if (pollData.response && pollData.response.generatedVideos && pollData.response.generatedVideos[0]) {
                    const videoBytes = pollData.response.generatedVideos[0].video.videoBytes;
                    const outPath = "/home/gateman/.openclaw/workspace/beauty_video.mp4";
                    fs.writeFileSync(outPath, Buffer.from(videoBytes, 'base64'));
                    console.log("Success! Video saved to:", outPath);
                } else {
                    console.log("Operation done but no video data:", JSON.stringify(pollData, null, 2));
                }
            }
        }
    }

  } catch (e) {
      console.error("Caught Exception:", e);
  }
}
main();

File: veo_test.py
import os
import time
from google import genai
from google.genai import types

os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "/home/gateman/sa-key.json"

try:
    print("Initializing GenAI client...")
    client = genai.Client(
        vertexai=True,
        project="jason-hsbc",
        location="us-central1"
    )

    print("Submitting video generation request to veo-3.1-generate-preview...")
    operation = client.models.generate_videos(
        model="veo-3.1-generate-preview",
        prompt="A highly professional and beautiful young female secretary confidently walking down a modern sunlit hallway, holding a sleek tablet. Cinematic lighting, ultra photorealistic, highly detailed, 4k.",
        config=types.GenerateVideosConfig(
            number_of_videos=1,
            aspect_ratio="16:9",
            person_generation="ALLOW_ADULT"
        )
    )

    print(f"Operation started: {operation.name}")
    
    op = operation
    attempts = 0
    while not op.done and attempts < 30:
        print(f"Waiting 10s... (Attempt {attempts+1})")
        time.sleep(10)
        # Re-fetch operation status using get
        op = client.operations.get(operation=op)
        attempts += 1

    if op.done:
        if op.error:
            print("Operation failed with error:", op.error)
        else:
            print("Operation completed!")
            video_bytes = op.response.generated_videos[0].video.video_bytes
            out_path = "/home/gateman/.openclaw/workspace/final_beauty_video.mp4"
            with open(out_path, "wb") as f:
                f.write(video_bytes)
            print(f"Success! Video saved to {out_path}")
    else:
        print("Timeout waiting for operation to complete.")

except Exception as e:
    print(f"Caught Exception: {e}")
